extensions/map-layers/src/ui/MapFeatureInfoTool.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import {
+  BeButtonEvent,
+  EventHandled,
+  HitDetail,
+  IModelApp,
+  LocateResponse,
+  PrimitiveTool,
+} from "@itwin/core-frontend";
+import { WidgetState } from "@itwin/appui-abstract";
+import { FrontstageManager, ToolItemDef } from "@itwin/appui-react";
+import { BeEvent } from "@itwin/core-bentley";
+import { FeatureInfoUiItemsProvider } from "./FeatureInfoUiItemsProvider";
+
+export const getDefaultMapFeatureInfoToolItemDef = (): ToolItemDef =>
+  new ToolItemDef({
+    toolId: DefaultMapFeatureInfoTool.toolId,
+    iconSpec: DefaultMapFeatureInfoTool.iconSpec,
+    label: () => DefaultMapFeatureInfoTool.flyover,
+    description: () => DefaultMapFeatureInfoTool.description,
+    execute: async () => { await IModelApp.tools.run(DefaultMapFeatureInfoTool.toolId); },
+  });
+
+export class DefaultMapFeatureInfoTool extends PrimitiveTool {
+  public static readonly onMapHit = new BeEvent<(hit: HitDetail) => void>();
+  public static override toolId = "MapFeatureInfoTool";
+  public static override iconSpec = "icon-map";
+
+  public override requireWriteableTarget(): boolean {
+    return false;
+  }
+
+  public override async onDataButtonDown(
+    ev: BeButtonEvent
+  ): Promise<EventHandled> {
+    const hit = await IModelApp.locateManager.doLocate(
+      new LocateResponse(),
+      true,
+      ev.point,
+      ev.viewport,
+      ev.inputSource
+    );
+    if (hit !== undefined) {
+      const widgetDef = FrontstageManager.findWidget(
+        FeatureInfoUiItemsProvider.widgetId
+      );
+      if (widgetDef && widgetDef.state !== WidgetState.Open)
+        widgetDef.setWidgetState(WidgetState.Open);
+
+      DefaultMapFeatureInfoTool.onMapHit.raiseEvent(hit);
```

Comment:
@mdastous-bentley , does it make more sense for users to define their own event and set MapLayersUi.onMapHit to that, or for us to define it and for them to just call `MapLayersUi.onMapHit.raiseEvent`? This is the only coupling left that im iffy about

Reply:
@aruniverse 
Yea I thought of that that,  I agree there is something weird about the API.  We can certainly 'own' the event, but I thought it would be confusing: generally when an object expose an event, it allows consumers to subscribe to this event and not fire it.  

So maybe we should keep this event internal, and expose a function/class with a method like `fetchFeatureInfoData(mapHit:HitDetails)`

Would it make more sense?

Reply:
Mind taking a look at the latest commit where i attempt to decouple the event from the singleton?

Reply:
@aruniverse added comments.

Reply:
discussed/tested and works fine; resolving comment